python/overlay_widget/shotgun_spinning_widget.py

```diff
@@ -0,0 +1,223 @@
+
+    def start_progress(self):
+        """
+        Enables the overlay and shows an animated progress arc.
+
+        If you want to stop the progress, call :meth:`hide`.
+        """
+        self.setVisible(True)
+        self._timer.start(1000 / self._UPDATES_PER_SECOND)
+        self._mode = self.MODE_PROGRESS
+        self._spin_angle_to = 0
+        self._spin_angle = 0
+
+    def report_progress(self, current):
+        """
+        Updates the widget current progress value.
+
+        :param current: New value for the progress arc. Must be between 0 (nothing) and 1 (complete).
+        :type current: float
+        """
+        # We're about to ask the cursor to reach another point. Make sure that
+        # we are at least caught up with where we were requested to be last time.
+        self._spin_angle = max(self._previous_spin_angle_to, self._spin_angle)
+
+        self._spin_angle_to = 360 * current
+        self.repaint()
+
+    def hide(self):
+        """
+        Hides the overlay.
+        """
+        self._timer.stop()
+        self._mode = self.MODE_OFF
+        self.setVisible(False)
+
+    ############################################################################################
+    # internal methods
+
+    def _on_animation(self):
+        """
+        Async callback to help animate the widget.
+        """
+        if self._mode == self.MODE_SPIN:
+            self._spin_angle += 1
+            if self._spin_angle == 90:
+                self._spin_angle = 0
+        elif self._mode == self.MODE_PROGRESS:
+            # If the current spin angle has not reached the destination yet,
+            # increment it, but not past where we are supposed to end at.
+
+            # The progress tries to give a smooth impression of the progress. Instead of jumping straight
+            # to the requested value, it will slide over to that value. Sliding from 0 to 1 however is done in
+            # a single second, so the sliding is still quick to the eye. If there are more than
+            # _UPDATES_PER_SECOND steps, this sliding effect is actually not visible since individual increments
+            # between steps will be smaller than 1 / _UPDATES_PER_SECOND of the circumference.
+            self._spin_angle = min(self._spin_angle_to, self._spin_angle + 360 / self._UPDATES_PER_SECOND)
+            self._heartbeat = (self._heartbeat + 1) % 25
+
+        self.repaint()
+
+    def _draw_opened_circle(self, painter, start_angle, span_angle):
+        """
+        Draws an arc around the SG logo.
+
+        :param painter: Painter object we will draw with.
+        :param start_angle: Angle at which we will start drawing the arc.
+        :param span_angle: Degrees the arc covers.
+        """
+        # show the spinner
+        painter.translate((painter.device().width() / 2) - 40,
+                          (painter.device().height() / 2) - 40)
+
+        pen = QtGui.QPen(QtGui.QColor("#424141"))
```

Comment:
Not sure what you mean. If you mean using an already existing constant, none actually match. The color is the same as the icon its drawing around, so the color is more tied to that icon than any core values.